File: Assignment9/C/Program1.c
// PS - Accept N number from user and return difference between summation of even elements and summation of odd elements

#include<stdio.h>
#include<stdlib.h>

int Arithmatic(int Arr[], int iLength){
    int iEven = 0;
    int iOdd = 0;
    int iCnt = 0;

    for(iCnt = 0; iCnt < iLength; iCnt++){
        if((Arr[iCnt] % 2) == 0) {
            iEven = iEven + Arr[iCnt];
        }
        if((Arr[iCnt] % 2) != 0) {
            iOdd = iOdd + Arr[iCnt];
        }
    }

    return (iEven - iOdd);
}

int main() {
    int iSize = 0;
    int *ptr = NULL;
    int iCnt =0;
    int iRet = 0;

    printf("Enter size of elements array \n");
    scanf("%d", &iSize);

    ptr = (int *)malloc(sizeof(int) * iSize);

    printf("Enter elements in array \n");
    for(iCnt = 0; iCnt < iSize; iCnt++){
        scanf("%d", &ptr[iCnt]);
    }

    iRet = Arithmatic(ptr, iSize);

    printf("Difference between summation of even and odd element is %d", iRet); 

    free(ptr);

    return 0;
}

File: Assignment34/Java/Program1.java
// PS - Write a java program which accept N numbers from user and return difference between summation of 
// even numbers and summation of oll elements

import java.util.*;

class ArrayX
{
    public int SumEven(int Arr[])
    {
        int iSum = 0;
        for(int iCnt = 0; iCnt< Arr.length; iCnt++)
        {
            if((Arr[iCnt] % 2) == 0)
            {
                iSum = iSum + Arr[iCnt];
            }
        }
        return iSum;
    }

    public int SumOdd(int Arr[])
    {
        int iSum = 0;
        for(int iCnt = 0; iCnt< Arr.length; iCnt++)
        {
            if((Arr[iCnt] % 2) != 0)
            {
                iSum = iSum + Arr[iCnt];
            }
        }
        return iSum;
    }

    public int Difference(int Arr[])
    {
        int Even = SumEven(Arr);
        int Odd = SumOdd(Arr);

        return Even - Odd;
    }
}

class Program1
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        int iSize = 0;
        System.out.println("Enter number of elements ");
        iSize = sobj.nextInt();

        int Arr[] = new int[iSize];

        System.out.println("Enter elements ");
        for(int iCnt = 0; iCnt < iSize; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }

        ArrayX aobj = new ArrayX();
        int iRet = aobj.Difference(Arr);

        System.out.println("Difference of even and odd numbers are : "+iRet);

    }
}